backend/nodes/properties/inputs/numeric_inputs.py

```diff
@@ -27,19 +38,32 @@ class NumberInput(BaseInput):
     def __init__(
         self,
         label: str,
-        default=0.0,
-        minimum=0,
-        maximum=None,
-        step=1,
-        optional=False,
-        number_type="any",
+        precision: int = 0,
+        offset: Union[float, int] = 0,
```

Comment:
This is problematic.

Let's use off numbers as an example. Since offset is 0 by default, odd numbers (min=some odd number, step=2) will not work. 

This is a pitfall that can be avoided if offset=minimum if minimum is defined.

Actually, maybe we could even get rid of this parameter completely if we say: `offset = minimum if minimum is not None else 0`.

Reply:
The whole point of the offset is you set it to 1 to enforce that numbers be odd. Right now if you set step to 2 and min to 1, it'll still let you type any even number

Reply:
But yeah I guess technically you could just derive that from the minimum value

Reply:
Makes sense, and was something else I'd been thinking about doing.

Reply:
> Actually, maybe we could even get rid of this parameter completely if we say: offset = minimum if minimum is not None else 0.

This actually doesn't work in a scenario where you have no minimum or maximum, but you still want odd numbers. Then minimum == None, but you still need offset to be 1.

Reply:
Since min=None effective means min=-∞, yes, offset is technically necessary.

However, are we ever going to have an input with an unbound range?

Reply:
> However, are we ever going to have an input with an unbound range?

Well that was my point. We can have full flexibility, or simplify for the range of scenarios we think we might reasonably encounter. We just need to pick which of those we want.

Reply:
I'm for simplifying. I don't think we will have much use for unbound ranges and if we did, we could approximate them by just choosing very large min-max values. offset can deal with large min-max values if we set it to `offset = min % step`, so there won't be any problems with floating point precision in the correction formula.

Reply:
Works for me.